components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowComponent.java

```diff
@@ -78,6 +87,7 @@
     @Metadata(label = "security")
     private String allowedRoles;
 
+    private UndertowSecurityProvider securityProvider;
```

Comment:
Add @Metadata with security label.

Reply:
@davsclaus this is not parameter which comes from configuration, it is just private variable that is used by component when SPI implementation is searched for. From my POV there should be annotation like @nometadata in tis case.